dashboard/src/components/Issue/IssueSection.tsx

```diff
@@ -0,0 +1,66 @@
+import { useMemo } from 'react';
+
+import { FaRegCircle, FaRegDotCircle } from 'react-icons/fa';
+
+import { FormattedMessage } from 'react-intl';
+
+import { UseQueryResult } from '@tanstack/react-query';
+
+import { Link } from '@tanstack/react-router';
+
+import { Skeleton } from '@/components/Skeleton';
+import ListingItem from '@/components/ListingItem/ListingItem';
+import { TIssue } from '@/types/general';
+
+import NoIssueFound from './NoIssueFound';
+
+const IssueSection = ({
+  data,
+  isLoading,
+  error,
+}: UseQueryResult<TIssue[]>): JSX.Element => {
+  const issueList = useMemo(
+    () =>
+      data?.map(issue => (
+        <Link
+          key={issue.incident_id}
+          to={issue.report_url}
+          target="_blank"
+          className="flex [&:not(:last-child)]:mb-2 [&:not(:last-child)]:border-b [&:not(:last-child)]:pb-2"
+        >
+          <ListingItem
+            text={issue.comment ?? ''}
+            leftIcon={
+              <div className="text-darkGray2">
+                {issue.present ? <FaRegDotCircle /> : <FaRegCircle />}
+              </div>
+            }
+          />
+        </Link>
+      )),
+    [data],
+  );
+
+  let sectionContent: React.ReactNode = issueList;
+
+  if (isLoading)
+    sectionContent = (
+      <Skeleton className="h-[100px]">
+        <FormattedMessage id="global.loading" />
+      </Skeleton>
+    );
+  else if (error)
+    sectionContent = <FormattedMessage id="issue.failedToFetch" />;
+  else if (!data || data.length === 0) sectionContent = <NoIssueFound />;
```

Comment:
You can use the `QuerySwitcher` component, I don't remember if there is support for empty data but you can add it there